includes/embeds/class-amp-base-embed-handler.php

```diff
@@ -151,4 +154,59 @@ protected function unwrap_p_element( DOMElement $node ) {
 			$parent_node->parentNode->replaceChild( $node, $parent_node );
 		}
 	}
+
+	/**
+	 * Removes the node's nearest `<script>` sibling with a `src` attribute containing the base `src` URL provided.
+	 *
+	 * @since 2.1
+	 *
+	 * @param DOMElement $node         The DOMNode to whose sibling is the script to be removed.
+	 * @param string     $base_src_url Script URL to match against.
+	 * @param string     $content      Text content of node to match against.
+	 */
+	protected function maybe_remove_script_sibling( DOMElement $node, $base_src_url, $content = '' ) {
```

Comment:
Here's an idea instead of passing `$base_src_url` and `$content`. What if instead this was passed a single `callable`? This callback function could then be passed the `DOMElement` as its argument, and then return a boolean for whether it matched. This would reduce some duplication in the method here and it would be more flexible for the matching logic, for example using regular expression matching. 

Also the `$content` arg isn't currently being used in the codebase.
```suggestion
	 * @param string     $base_src_url Script URL to match against.
	 * @param callable   $match_callback Callback which is passed the script element to determine if it is a match.
	 */
	protected function maybe_remove_script_sibling( DOMElement $node, $match_callback ) {
```

Reply:
That could work. Done in https://github.com/ampproject/amp-wp/pull/5926/commits/c33059703a54652e06dcb2cc403c9501e10a1ce8.